sql/sql_table.cc

```diff
@@ -6559,16 +6559,18 @@ Compare_keys compare_keys_but_name(const KEY *table_key, const KEY *new_key,
                       ((Field_varstring *) old_field)->length_bytes);
     }
 
+    auto is_equal= key_part->field->is_equal(new_field);
```

Comment:
We might want to cherry-pick this into an earlier major version where C++11 is not allowed. Hence, it might be better to fill in the actual type instead of letting `auto` infer it.